salt/edx/tests/test_edxapp_running.sls

```diff
@@ -0,0 +1,86 @@
+#!jinja|yaml
+
+{# Test to verify that the edxapp is funcitional and configured with some of
+our configurations. Test the following:
+- edxapp supervisor services are running
+- edxapp instance can reach select services
+- services that edxapp relies on locally
+- edxapp:cms service is only running on edx-draft
+- AWS EBS data repo is mounted
+- spot check custom vars are in lms.{env|auth} #}
+
+
+{% set supervisor_services = [
+    'edxapp:lms',
+    'forum',
+    'xqueue',
+    'xqueue_consumer'
+  ] %}
+
+{% set socket_connections = {
+    'ssl': 'tcp://0.0.0.0:443',
+    'lms': 'tcp://0.0.0.0:8000',
+    'gitreload_service': 'tcp://0.0.0.0:8095',
+    'xqueue': 'tcp://0.0.0.0:18040',
+    'forum': 'unix:///edx/var/forum/forum.sock',
+    'rabbitmq': 'tcp://rabbitmq.service.consul:15672',
+    'elasticsearch': 'tcp://elasticsearch.service.consul:9200',
+    'mysql':'tcp://mysql.service.consul:3306',
+    'mongodb': 'tcp://mongodb-master.service.consul:27017',
+  } %}
+
+{% set running_services = [
+    'nginx',
+    'fluentd',
+    'supervisor',
+    'gitreload'
+  ] %}
+
+{% set lms_env = {
+  'mitx_email': salt.pillar.get('edx:ansible_vars:edxapp_generic_env_config:DEFAULT_FROM_EMAIL'),
+  'mitx_theme': salt.pillar.get('edx:ansible_vars:edxapp_generic_env_config:THEME_NAME'),
+  'rabbitmq_service': salt.pillar.get('edx:ansible_vars:xqueue_env_config:RABBIT_HOST')
+  } %}
+
+{% for sv_service in supervisor_services %}
+test_edxapp_supervisor_{{ sv_service }}:
+  testinfra.supervisor:
+    - name: {{ sv_service }}
+    - is_running: True
+{% endfor %}
+
+{% for connection in socket_connections %}
+test_edxapp_{{ connection }}:
+  testinfra.sockethost:
+    - name: {{ connection }}
+    - is_listening: True
+{% endfor %}
+
+{% for service in running_services %}
+test_edxapp_{{ service }}:
+  testinfra.service:
+    - name: {{ service }}
+    - is_running: True
+{% endfor %}
+
+test_edxapp_cms:
+  testinfra.supervisor:
+      - name: 'edxapp:cms'
+      - is_running: True
+
+# Check if AWS EFS is mounted
+test_edxapp_efs_mount:
+  testinfra.mountpoint:
+    - name: '/mnt/data'
+    - filesystem:
+        expected: nfs4
+        comparison: eq
+
+{% for entry in lms_env %}
+test_edxapp_lms_env_{{ entry }}:
+  testinfra.file:
+    - name: '/edx/app/edxapp/lms.env.json'
+    - contains: {{ entry }}
```

Comment:
The `for` loop should be more like `for attribute, value in lms_env.items()`. Just doing a for loop through the dictionary will only give you the keys and not the values.

The `contains` entry should look like:
```salt
- contains:
  parameter: {{ value }}
  expected: True
  comparison: is_
```
See https://github.com/mitodl/rabbitmq-formula/blob/master/rabbitmq/tests/test_configure.sls#L11 for an example.